### PR TITLE
Feature/diffuse material

### PR DESCRIPTION
**Pull request by andrearastelli**
There is an issue that must be fixed with the sphere hit process.

The problem is obviously generated in this PR.

test_diffuse

### CODE REVIEW

**Comment by andrearastelli** [comment]
Also, the random_in_sphere point generation function, right now uses the `drand48()` function, that is not compatible with Visual Studio.

Must figure out a way to generate random numbers for this function faster than with `std::uniform_real_distribution`.

File: src/color.h [hunk 1]
```diff
@@ -89,6 +91,16 @@ Color& Color::operator/=(float a)
 }
 
 
+Color& Color::gamma()
```
Comment (andrearastelli) [inline src/color.h]: Very stupid gamma function to liearize colors.

File: src/main.cpp [hunk 2]
```diff
@@ -9,62 +9,85 @@
 #include "hitablelist.h"
 #include "camera.h"
 
+std::random_device d;
+std::mt19937 m{d()};
+std::uniform_real_distribution<float> dist(0.0f, 1.0f);
```
Comment (andrearastelli) [inline src/main.cpp]: Global variables for random number processing.
>This should be fixed, removing the global variables from the main.cpp

File: src/main.cpp [hunk 3]
```diff
@@ -9,62 +9,85 @@
...
-    Hitable *world = new HitableList(list, 2);
+    Hitable *world = new HitableList(list, i);
 
     Camera cam;
 
     // RANDOM GENERATORS
     std::random_device d;
     std::mt19937 m{d()};
-    std::uniform_real_distribution<float> dist(0.0f, 1.0f);
+    auto max_rand_jitter = 1.0f - 1.0f / samples;
```
Comment (andrearastelli) [inline src/main.cpp]: The sample "radius" is processed as:
`1 - (1 / samples)`
This way, with only 1 sample, the random function will return 0, always, for each pixel (this way no jitter is generated).
With more than 1 sample, with higher values the random radius will approach 1.0, generating more smooth images, without having huge gaps between samples.

File: src/color.h (hunk 1 above)
Comment (Ale32) [reply src/color.h]: emoji even here...

File: src/main.cpp [hunk 4]
```diff
@@ -9,23 +9,29 @@
...
+
 Vec3 random_in_unit_sphere();
 
 Color ray_color(const Ray &r, Hitable *world);
 
 
 int main()
 {
-    Image i("test_diffuse.ppm");
+    Image image("test_diffuse.ppm");
```
Comment (Ale32) [inline src/main.cpp]: Much nicer

File: src/main.cpp [hunk 5]
```diff
@@ -9,23 +9,29 @@
...
 
     auto samples = 100;
 
-    Hitable *list[2];
+    Hitable *list[3];
 
-    list[0] = new Sphere(Vec3(0.0f, 0.0f, -1.0f), 0.5f);
-    list[1] = new Sphere(Vec3(0.0f, -5.0f, 0.0f), 5.0f);
+    int i = 0;
```
Comment (Ale32) [inline src/main.cpp]: Oh ok, that's why you turned `i` into `image`

Comment (andrearastelli) [reply src/main.cpp]: `int list_idx` was too long... :)

**Comment by andrearastelli** [comment]
@Ale32 why merging `develop` into this branch?
What happened?

**Pull request merged**